### PR TITLE
Owl assessment criteria

### PR DESCRIPTION
**Pull request by adrianclay**

### CODE REVIEW

**Comment by danielburnley** [comment]
Rendered version [here](https://github.com/madetech/learn/blob/b6834c2c890b64338cfbe4afa5052e2fd0406491/core-skills/web-application-development-with-react/README.md)

File: core-skills/web-application-development-with-react/README.md [hunk 1]
```diff
@@ -92,29 +92,44 @@ During the assessment, the reviewer will go through your application with you as
...
-**Coming Soon - Owl is intended to demonstrate ability to put understanding of advanced architecture
-principles and testing strategy into practice**
+#### Getting wise with React - Integrating with remote services
+
+For this level you are expected to demonstrate the ability to test both your React 
+components and any business logic used for communicating with remote services. It will
+also cover demonstrating best practice around the separation of business logic and UI
+logic.
+
+During the asessement, your reviewer will go through your application asking for
```
Comment (zuren) [inline core-skills/web-application-development-with-react/README.md]: _assessment_

File: core-skills/web-application-development-with-react/README.md [hunk 2]
```diff
@@ -92,29 +92,44 @@ During the assessment, the reviewer will go through your application with you as
...
+  - Candidate should walk assessor through tests which give maintainers confidence 
+    that the component submits the data from the form to the API.
+- There should be a JS layer of indirection between the components and the API
+  - Candidate should showcase a class which makes HTTP API requests, and tests which
+    cover this.
+  - Candidate should showcase tests for the component which mocks boundary between the
+    component and model.
+- The application should have component tests should have minimal coupling to the HTML
+  structure they render.
+  - Candidate should showcase their use of test specific attributes in the test suite
```
[the remark this answers is not in the document]
Comment (zuren) [reply core-skills/web-application-development-with-react/README.md]: On this - terms we use in the marking guide should probably be clearly explained in the learning materials.

File: core-skills/web-application-development-with-react/README.md [hunk 3]
```diff
@@ -125,11 +125,9 @@ The candidate should create a React application which satisfies the following.
...
-  - Candidate should showcase a class which makes HTTP API requests, and tests which
+  - Candidate should showcase standalone code which makes HTTP API requests, and tests which
     cover this.
-  - Candidate should showcase tests for the component which mocks boundary between the
-    component and model.
-- The application should have component tests should have minimal coupling to the HTML
-  structure they render.
-  - Candidate should showcase their use of test specific attributes in the test suite
+  - Candidate should showcase tests which prove minimal coupling between the component and API interface.
+- The application should have component tests should have minimal coupling to markup they render.
```
Comment (zuren) [inline core-skills/web-application-development-with-react/README.md]: "_should_ have component tests _should_ have minimal coupling to markup they render"

Dan this makes no sense.

File: core-skills/web-application-development-with-react/README.md [hunk 4]
```diff
@@ -92,29 +92,42 @@ During the assessment, the reviewer will go through your application with you as
...
+The candidate should create a React application which satisfies the following.
+
+- Has a component which displays data to the user which has been fetched from an API
+  - Candidate should showcase their component to assessor.
+  - Candidate should walk assessor through tests which give maintainers confidence
+    that the component displays data from an API.
+- Has a form component which takes information from the user and submits said information
+  to an API.
+  - Candidate should showcase their form to assessor.
+  - Candidate should demonstrate that the API has recieved information successfully
```
Comment (StevenLeighton21) [inline core-skills/web-application-development-with-react/README.md]: Can this be via a test? Does the API have to exist?

Comment (adrianclay) [reply core-skills/web-application-development-with-react/README.md]: In Dans case he wasn't using an API which had any POST endpoints, it was purely GET.  He achieved this by:

A) Having an automated test.
B) Using the network tools to show that a POST had been made.

Comment (StevenLeighton21) [reply core-skills/web-application-development-with-react/README.md]: Reading back my comments don't seem very clear - just seeking to clarify if an automated test mocking an API with a POST endpoint suffices to prove an API receives information.

Comment (adrianclay) [reply core-skills/web-application-development-with-react/README.md]: I will make this criterion more specific:

- Candidate should open the network tools and show that on form submission, a request was made by the browser to the API endpoint with a JSON encoded response body containing the form data.

Comment (danielburnley) [reply core-skills/web-application-development-with-react/README.md]: Yeah showing the json request would probably be a good thing

**Pull request merged**